service/src/main/java/bio/terra/workspace/service/resource/controlled/flight/clone/azure/database/DumpAzureDatabaseStep.java

```diff
@@ -0,0 +1,156 @@
+import bio.terra.workspace.service.resource.controlled.cloud.azure.database.ControlledAzureDatabaseResource;
+import bio.terra.workspace.service.resource.controlled.cloud.azure.storageContainer.ControlledAzureStorageContainerResource;
+import bio.terra.workspace.service.resource.controlled.model.AccessScopeType;
+import bio.terra.workspace.service.resource.model.StewardshipType;
+import bio.terra.workspace.service.resource.model.WsmResourceFamily;
+import bio.terra.workspace.service.resource.model.WsmResourceType;
+import bio.terra.workspace.service.workspace.WorkspaceService;
+import bio.terra.workspace.service.workspace.flight.WorkspaceFlightMapKeys;
+import bio.terra.workspace.service.workspace.model.AzureCloudContext;
+import java.text.SimpleDateFormat;
+import java.util.UUID;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class DumpAzureDatabaseStep implements Step {
+  private static final Logger logger = LoggerFactory.getLogger(DumpAzureDatabaseStep.class);
+
+  private final ControlledAzureDatabaseResource sourceDatabase;
+  private final LandingZoneApiDispatch landingZoneApiDispatch;
+  private final SamService samService;
+  private final WorkspaceService workspaceService;
+  private final AzureStorageAccessService azureStorageAccessService;
+  private final AzureDatabaseUtilsRunner azureDatabaseUtilsRunner;
+  private final WsmResourceService wsmResourceService;
+
+  public DumpAzureDatabaseStep(
+      ControlledAzureDatabaseResource sourceDatabase,
+      LandingZoneApiDispatch landingZoneApiDispatch,
+      SamService samService,
+      WorkspaceService workspaceService,
+      AzureStorageAccessService azureStorageAccessService,
+      AzureDatabaseUtilsRunner azureDatabaseUtilsRunner,
+      WsmResourceService wsmResourceService) {
+
+    this.sourceDatabase = sourceDatabase;
+    this.landingZoneApiDispatch = landingZoneApiDispatch;
+    this.samService = samService;
+    this.workspaceService = workspaceService;
+    this.azureStorageAccessService = azureStorageAccessService;
+    this.azureDatabaseUtilsRunner = azureDatabaseUtilsRunner;
+    this.wsmResourceService = wsmResourceService;
+  }
+
+  @Override
+  public StepResult doStep(FlightContext context) throws InterruptedException, RetryException {
+    var inputParameters = context.getInputParameters();
+    var workingMap = context.getWorkingMap();
+    var destinationWorkspaceId =
+        inputParameters.get(
+            WorkspaceFlightMapKeys.ControlledResourceKeys.DESTINATION_WORKSPACE_ID, UUID.class);
+    var userRequest =
+        getRequired(
+            inputParameters,
+            JobMapKeys.AUTH_USER_INFO.getKeyName(),
+            AuthenticatedUserRequest.class);
+
+    // TODO WM-2366: replace with storage container created in WM-2371
+    ControlledAzureStorageContainerResource destinationContainer =
+        wsmResourceService
+            .enumerateResources(
```

Comment:
perhaps the [enumerateResources in ResourceDao](https://github.com/DataBiosphere/terra-workspace-manager/blob/ad7ebaffadb34f5d87291de438ecaec34e8799c5/service/src/main/java/bio/terra/workspace/db/ResourceDao.java#L269) would be better